sdk/scala-sdk-testkit/src/main/scala/kalix/scalasdk/testkit/impl/TestKitTimerScheduler.scala

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package kalix.scalasdk.testkit.impl
+
+import scala.concurrent.Future
+import scala.concurrent.duration.FiniteDuration
+
+import akka.Done
+import kalix.javasdk.impl.DeferredCallImpl
+import kalix.scalasdk.DeferredCall
+import kalix.scalasdk.impl.ScalaDeferredCallAdapter
+import kalix.scalasdk.testkit.SingleTimerDetails
+import kalix.scalasdk.timer.TimerScheduler
+
+class TestKitTimerScheduler extends TimerScheduler {
+
+  private var _singleTimers: Seq[SingleTimerDetails] = Seq.empty
+  private var _timerCancellations: Seq[String] = Seq.empty
```

Comment:
Probably needs to be `@volatile` since mutable and _maybe_ written to from future/cs callbacks (even if the opposite, set timer, do async stuff, is probably more common)